### PR TITLE
`Exam mode`: Fix translations

### PR DESCRIPTION
**Pull request by b-fein**
#### General

- [ ] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/).
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context
Exam translations were pretty broken because the JSON structure did not match the keys used in the code. Additionally, the create test runs dialogue was missing translations.

### Description
This PR fixes the structure of the JSON files and adds translation to the dialogue.

### Steps for Testing
1. Create an exam as instructor and check the various exam management pages for translation errors.

#### Code Review (when the translations are shown correctly, then the code changes worked)
- [ ] Review 1
- [ ] Review 2
#### Manual Tests
- [x] Test 1
- [ ] Test 2

### Test Coverage
unchanged

### Screenshots
<details>
<summary>Click to Unfold</summary>

Before:
Screenshot_exam_before
create_test_run_before

After:
Screenshot_exam_after
test_run
</details>

### CODE REVIEW

**Review by cremertim — APPROVED** [review]
translation looks good to me.  Great job 👍

**Review by baranimo7 — COMMENTED** [review]
Everything works as expected. There is no ,,Translation not found" message.

**Review by MertAksehirlioglu — APPROVED** [review]
Tested on TS1 and translations look as expected

File: src/main/webapp/i18n/de/exam.json [hunk 1]
```diff
@@ -313,70 +315,32 @@
...
-                "publishResults": "Veröffentlichen der Ergebnisse",
-                "examReview": "Planung der Klausureinsicht ",
-                "resolveComplaints": "Auflösen aller Beschwerden",
-                "testRun": "Abhalten eines Testdurchlaufs",
-                "exportResults": "Exportiere die Klausurergebnisse"
+            "exerciseGroups": "Aufgabengruppen",
+            "exerciseGroup": {
+                "create": "Neue Aufgabengruppe erstellen",
+                "update": "Aufgabengruppe bearbeiten",
+                "type": "Type",
```
Comment (b-fein) [inline src/main/webapp/i18n/de/exam.json:323]:
```suggestion
                "type": "Typ",
```

**Review by Strohgelaender — CHANGES_REQUESTED** [review]
tested in testing session with @lBruchmann @mallgaier @cremertim. The changes worked as expected, but we noticed some further inconsistencies among the translations that should be addressed:

- The German translations mix the usage of "Sie" and "Du", we should be consistent
- exam checklist item 1: "Aufgaben Gruppen" should be "Aufgabengruppe"
- The translations mix nominal style (e.g on the checklist "Auflösen aller Beschwerden" and "Exportiere die Klausurergebnisse")

**Review by mallgaier — CHANGES_REQUESTED** [review]
Tested together with @Strohgelaender  and @lBruchmann . For our suggestions see the contribution from @Strohgelaender

**Comment by b-fein** [comment]
Thanks for the reviews!
- Most places use the familiar ‘du’, so I replaced the remaining usages of ‘Sie’.
- The compound-nouns with spaces have been fixed. 
- Two more places with a missing translation have been fixed (last two commits).

The exam checklist has been updated:
<details>
<summary>Click to unfold screenshot</summary>

Screenshot 2021-11-24 at 06-22-22 Klausurverwaltung
</details>

File: src/main/webapp/i18n/de/exam.json [hunk 2]
```diff
@@ -313,98 +315,60 @@
...
             },
-
             "reset": {
                 "question": "Soll die Klausur <strong>{{ title }}</strong> wirklich dauerhaft zurückgesetzt werden? Alle Klausuren der Studierenden werden gelöscht. Teilnehmende an der Klausur und die Konfigurationen der Aufgaben bleiben erhalten. Diese Aktion kann NICHT rückgängig gemacht werden!",
                 "typeNameToConfirm": "Bitte gib den Namen der Klausur zur Bestätigung ein."
             },
             "checklist": {
                 "title": "Checkliste für die Durchführung von Prüfungen",
-                "introduction": "Unter folgendem Link kannst du dich über den Ablauf einer Artemis Klausur informieren:",
+                "introduction": "Unter folgendem Link kannst du dich über den Ablauf einer Klausur in Artemis informieren:",
```
Comment (Strohgelaender) [inline src/main/webapp/i18n/de/exam.json:350]:
```suggestion
                "introduction": "Unter folgendem Link kannst du dich über den Ablauf einer Klausur auf Artemis informieren:",
```

**Review by Strohgelaender — CHANGES_REQUESTED** [review]
Thanks for addressing this. A few more points:

-  Can you also translate the link to the Instructors Guide
grafik
- Test Run Management: "Anschließend, kannst du die Abgabe als Tutor:in bewerten." -> "Anschließend kannst du die Abgabe als Tutor:in bewerten." (delete the komma, or maybe even change the whole text: "Anschließend können Tutor:innen die Abgabe bewerten. Das Ergebnis könnnen Studierende einsehen und sich beschweren. Diese Beschwerden können dann Tutor:innen bewerten.")
- The exercise group overview still contains a missing translation ("Number of exercise groups"):
grafik
- The Exam Assessment Dashboard has some grammar errors:
grafik
- The "assessments" header for the tutor leaderboard is not translated:
grafik
- Student Exams: Missing translation for the "Generate missing individual exams" button (there is no corresponding string in the de/exam.json file)
grafik

**Comment by b-fein** [comment]
Thanks for the re-review! I applied the changes.

> Test Run Management: "Anschließend, kannst du die Abgabe als Tutor:in bewerten."

I only removed the comma, as the intention would change otherwise as far as I understand. The test run is for an instructor to go through all phases of the exam by themself while looking at the test submission from the perspective like e.g. a student would be able to see it. Therefore, I kept the various ‘[…] kannst du […]’ parts.

checklist_intro
exercise_group_management
exam_assessment_dashboard
student_exam_management

File: src/main/webapp/i18n/de/exam.json (hunk 2 above)
Comment (b-fein) [reply src/main/webapp/i18n/de/exam.json]: The feature overview page uses both ‘auf’ and ‘in’ for ‘Onlineklausuren in/auf Artemis’. I personally favour ‘in’, but also don’t want to just dismiss your suggestion as both variants are perfectly valid ways to phrase this.

**Comment by Strohgelaender** [comment]
> The test run is for an instructor to go through all phases of the exam by themself while looking at the test submission from the perspective like e.g. a student would be able to see it. Therefore, I kept the various ‘[…] kannst du […]’ parts.

That makes sens, thank you for the explanation and the fixes 👍

**Review by Strohgelaender — DISMISSED** [review]
tested again, could not find any further problems.

**Review by mallgaier — CHANGES_REQUESTED** [review]
Great work, the mentioned changes are implemented so far. 
The button on the "exam management" page to create an exam is still in English. 
image
Apart from the smaller issue, I did not notice any more errors when I looked through the exam page.

**Pull request merged**